File: data/1002/F_h.java
/* Copyright Howard Cheng (ACM ICPC ECNA 1999) */

import java.io.*;
import java.util.*;

public class F_h {

  public static void main(String [] args) {
    DataInputStream in = new DataInputStream(System.in);
    String line = "";
    int case_num = 1, k;

    try {
      line = in.readLine();
    } catch (IOException ioe) {}
    StringTokenizer str = new StringTokenizer(line);
    int p = Integer.parseInt(str.nextToken());
    int e = Integer.parseInt(str.nextToken());
    int i = Integer.parseInt(str.nextToken());
    int d = Integer.parseInt(str.nextToken());
      
    while (p != -1 && e != -1 && i != -1 && d != -1) {
      for (k = d+1; (k - p) % 23 != 0 ||
	            (k - e) % 28 != 0 ||
                    (k - i) % 33 != 0; k++)
	;
      System.out.println("Case " + (case_num++) + 
			 ": the next triple peak occurs in " +
			 (k - d) + " days.");
      try {
	line = in.readLine();
      } catch (IOException ioe) {}
      str = new StringTokenizer(line);
      p = Integer.parseInt(str.nextToken());
      e = Integer.parseInt(str.nextToken());
      i = Integer.parseInt(str.nextToken());
      d = Integer.parseInt(str.nextToken());
    }
  }
}

File: data/1002/F-H.c
/* Copyright Howard Cheng (ACM ICPC ECNA 1999) */

#include <stdio.h>
#include <stdlib.h>
#include <assert.h>

int main(void)
{
  int d, a[3], m[3], M, i, case_num;

  m[0] = 23;
  m[1] = 28;
  m[2] = 33;
  M = 23*28*33;

  for (case_num = 1; 
       scanf("%d %d %d %d", a, a+1, a+2, &d) == 4 && 
	 a[0] != -1 && a[1] != -1 && a[2] != -1 && d != -1;
       case_num++) {
    for (i = d+1; (i-a[0]) % m[0] || (i-a[1]) % m[1] || (i-a[2]) % m[2]; i++)
      ;

    printf("Case %d: the next triple peak occurs in %d days.\n", case_num,
	   i-d);
  }

  return 0;
}

